src/testing/jest/jest-preprocessor.ts

```diff
@@ -220,7 +238,7 @@ function getCompilerOptions(rootDir: string): ts.CompilerOptions | null {
  * @returns `true` if the file should be transformed, `false` otherwise
  */
 export function shouldTransform(filePath: string, sourceText: string): boolean {
-  const ext = filePath.split('.').pop().toLowerCase().split('?')[0];
+  const ext = (filePath.split('.').pop() ?? '').toLowerCase().split('?')[0];
```

Comment:
does this actually remove an error? in a node repl if I do

```js
"".split(".").pop()
```

I get `""`, so I'm not sure the `?? ""` gives us additional safety here. if I'm reading the [ECMA standard](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.split) correctly if the substring to split on isn't found in the string to split then the return value should be a singleton array containing the original string (unless the original string is `""` _and_ the string / regexp on which to split matches that)

Reply:
I think you're correct, the problem isn't ECMA but how TypeScript types `pop()` here:

```
   /**
     * Removes the last element from an array and returns it.
     * If the array is empty, undefined is returned and the array is not modified.
     */
    pop(): T | undefined;
```

Since TypeScript can't infer that the result of `split()` will always be a non-empty array

without the `?? ''`, we get:

```
src/testing/jest/jest-preprocessor.ts:241:15 - error TS2532: Object is possibly 'undefined'.

241   const ext = (filePath.split('.').pop()).toLowerCase().split('?')[0];
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Sadly we add verbosity here because of TS 😢

Reply:
typescript! \*shakes fist\*

alright sounds good :)